bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -212,3 +390,188 @@ def classify_columns(
         metrics_sorted,
         undefined_sorted,
     )
+
+
+def generate_query_with_shredder_mitigation(
+    client, project_id, dataset, destination_table, backfill_date=PREVIOUS_DATE
+) -> Path:
+    """Generate a query to backfill with shredder mitigation."""
+
+    # Find query files and grouping of previous and new 'current' query.
+    new = Subset(
+        client, destination_table, None, "new_version", dataset, project_id, None
+    )
+    previous = Subset(
+        client, new.name_previous_version, None, "previous", dataset, project_id, None
+    )
+
+    query_with_mitigation_name = "query_with_shredder_mitigation"
+    query_with_mitigation_path = Path("sql") / project_id
+
+    # Check that previous query exists and GROUP BYs are valid in both queries.
+    previous_group_by = extract_last_group_by_from_query(previous.get_query_path())
+    new_group_by = extract_last_group_by_from_query(new.get_query_path())
+    if (
+        "ALL" in previous_group_by
+        or "ALL" in new_group_by
+        or not all(isinstance(e, str) for e in previous_group_by)
+        or not all(isinstance(e, str) for e in new_group_by)
+        or not previous_group_by
+        or not new_group_by
+    ):
```

Comment:
should you also avoid `GROUP BY 1, 2` here?

Reply:
Definitely, I added a validation, thanks!